apps/accounts/models/user.py

```diff
@@ -86,6 +79,63 @@ def __str__(self):
     def is_admin(self):
         return self.groups.filter(name="admin").exists()
 
+    @property
+    def hosting_providers(self) -> models.QuerySet[Hostingprovider]:
+        """
+        Returns a QuerySet of all Hostingproviders that the User has permissions to manage.
+
+        This method is useful to verify whether a user has permission to manage the hosting provider
+        """
+        return get_objects_for_user(self, str(manage_provider))
+
+    @property
+    def hosting_providers_explicit_perms(self) -> models.QuerySet[Hostingprovider]:
+        """
+        Returns a QuerySet of all Hostingproviders that the User has *explicit* permissions to manage,
+        not taking into consideration:
+            - global permissions
+            - group membership
+            - superuser status
+
+        This method is useful to fetch hosting providers for users that belong to the admin group.
+        """
+        return get_objects_for_user(
+            self,
+            str(manage_provider),
+            accept_global_perms=False,
+            with_superuser=False,
+            use_groups=False,
+        )
+
+    @property
+    def data_centers(self) -> models.QuerySet[Datacenter]:
+        """
+        Returns a QuerySet of all Datacenters that the User has permissions to manage.
+
+        This method is useful to verify whether a user has permission to manage the datacenter.
+
+        """
+        return get_objects_for_user(self, str(manage_datacenter))
+
+    @property
+    def data_centers_explicit_perms(self) -> models.QuerySet[Datacenter]:
```

Comment:
As above with the comments about why we have an explicit method. I suspect it won't be clear why we have to similar methods in a few months, and it would be nice to be kind to our future selves.